services/bettersocial/fcmToken/sendGroupChatNotification.js

```diff
@@ -0,0 +1,26 @@
+const {messaging} = require('firebase-admin');
+const FcmTokenFunction = require('../../../databases/functions/fcmToken');
+const {FcmToken} = require('../../../databases/models');
+
+const sendGroupChatNotification = async (userTargetId, message) => {
+  const userTargetToken = await FcmTokenFunction.findTokenByUserId(FcmToken, userTargetId);
+
+  const payload = {
+    notification: {
+      title: message,
+      body: message
+    },
+    data: {}
+  };
+  if (userTargetToken) {
+    try {
+      messaging()
+        .sendToDevice(userTargetToken?.token, payload)
+        .then(() => {});
+    } catch (error) {
+      console.log('error send group chat notifications', error);
+    }
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a potential issue with error handling here. The `try-catch` block won't catch any errors thrown by `messaging().sendToDevice()` because it returns a Promise and the error will be thrown asynchronously. To fix this, you should `await` the call to `messaging().sendToDevice()`. Also, the `.then(() => {})` is unnecessary and can be removed.

```diff
-      try {
-        messaging()
-          .sendToDevice(userTargetToken?.token, payload)
-          .then(() => {});
-      } catch (error) {
-        console.log('error send group chat notifications', error);
-      }
+      try {
+        await messaging().sendToDevice(userTargetToken?.token, payload);
+      } catch (error) {
+        console.log('error send group chat notifications', error);
+      }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->